### PR TITLE
Simplify duplicated code and add moveMotorStep

### PR DESCRIPTION
**Pull request by Depau**
If you prefer the new movement method and the de-duplication as separate pull requests, let me know. It's simpler for me to send everything in one batch, though :)

This pull request, other than adding `motorMoveStep` which moves the motor in a direction, removes all the duplicated request handling code and puts it into a common helper method.

The URL generation was rewritten using `str.format`. Also I removed `:443` since 443 is the standard HTTPS port.

I removed the `raiseException` parameter: it is only useful if the session token has expired, so I moved that part in the common `performRequest` method and the token is refreshed only if the camera returned a stok error (otherwise an exception is always thrown).

### CODE REVIEW

File: pytapo/__init__.py [hunk 1]
```diff
@@ -71,73 +73,71 @@ def responseIsOK(self, res):
...
-        data = json.loads(res.text)
-        if self.responseIsOK(res):
-            return json.loads(res.text)
-        else:
-            if raiseException:
+        data = res.json()
+        if not self.responseIsOK(res):
+            if data["error_code"] == -40401 and loginRetryCountdown > 0:  # Invalid Stok
+                self.refreshStok()
+                return self.performRequest(inData, loginRetryCountdown - 1)
```
Comment (JurajNyiri) [inline pytapo/__init__.py:91]: What is the reason of 5 retries?

File: pytapo/__init__.py [hunk 2]
```diff
@@ -71,73 +73,71 @@ def responseIsOK(self, res):
...
-            url, data=json.dumps(data), headers=self.headers, verify=False
+            url, data=json.dumps(inData), headers=self.headers, verify=False
         )
-        data = json.loads(res.text)
-        if self.responseIsOK(res):
-            return json.loads(res.text)
-        else:
-            if raiseException:
+        data = res.json()
+        if not self.responseIsOK(res):
```
Comment (JurajNyiri) [inline pytapo/__init__.py:88]: I am not sure how res.json behaves if the response is not a json. Could we move `data = res.json()` after if not self.responseIsOK(res): instead?

**Review by JurajNyiri — COMMENTED** [review]
This is awesome! Thank you 👍 

Before approval and deploy, I will need to test the functionality - I will get to it in upcoming days.

Had some comments / questions below.

**Comment by JurajNyiri** [comment]
Lets use res.json also in this line:
`self.stok = json.loads(res.text)["result"]["stok"]`

File: pytapo/__init__.py (hunk 1 above)
Comment (Depau) [reply pytapo/__init__.py:91]: Actually none, I can set it to 1 :)

File: pytapo/__init__.py (hunk 2 above)
Comment (Depau) [reply pytapo/__init__.py:88]: I see

**Comment by Depau** [comment]
Both comments resolved, I lowered the login retry count to 1 and I'm only doing `json.loads` when `not responseIsOk`. Everywhere else it's `res.json()`, since even in `responseIsOk` such an issue woulld be wrapped in another exception anyway.

**Comment by JurajNyiri** [comment]
Thank you for your contribution! I will deploy this after I finish testing and adding unit tests.

**Pull request merged**